drivers/net/phy/marvell10g_ptp.c

```diff
@@ -473,3 +601,254 @@ static long mv3310_do_aux_work(struct ptp_clock_info *ptp)
+	    (ig_parser_check != MV_V2_PTP_UDATA_EMPTY) &&
+	    (ig_updater_check != MV_V2_PTP_UDATA_EMPTY))
+		return 0;
+
+	dev_info(&phydev->mdio.dev, "loading PTP parser & updater microcode\n");
+	return mv3310_ptp_load_ucode(priv);
+}
+
+static int mv3310_ptp_set_lut(struct phy_device *phydev)
+{
+	int ret;
+
+	/* Set Egress (TX) LUT Match Key..
+	 * TRANSPORTSPECIFIC   MESSAGETYPE  VERSIONPTP ...(zeros)... FLAGPTPV2
+	 *      0000              0000      0000 0010                    1
+	 *      PTPv2        Sync/Delay_Req     2                      PTPv2
+	 * Sync = 0000, Delay_Req = 0001 => MESSAGETYPE (value) = 000* (use 0 as *).
+	 * Ignore FLAGFIELD, DOMAINNUMBER. */
+	ret = mv3310_write_ptp_lut_reg(phydev, MV_V2_PTP_LUT_KEY_EG_BASE,
+				       0x00020001);
+	if (ret < 0)
+		return ret;
+
+	/* Set Egress (TX) LUT Match Enable. This is mask. Set to 1 bit positions
+	 * from Egress (TX) LUT Match Key above..
+	 * Check TRANSPORTSPECIFIC, MESSAGETYPE, VERSIONPTP and FLAGPTPV2:
+	 * TRANSPORTSPECIFIC   MESSAGETYPE  VERSIONPTP ...(zeros)... FLAGPTPV2
+	 *      1111               1110      0000 1111                    1
+	 *      PTPv2        Sync/Delay_Req      2                      PTPv2
+	 * Sync = 0000, Delay_Req = 0001 => MESSAGETYPE (mask) = 1110. */
+	ret = mv3310_write_ptp_lut_reg(phydev, MV_V2_PTP_LUT_KEY_EG_BASE + 8,
+				       0xfe0f0001);
+	if (ret < 0)
+		return ret;
+
+	/* Set Ingress (RX) LUT Match Key (what to compare with).
+	 * TRANSPORTSPECIFIC   MESSAGETYPE   VERSIONPTP ...(zeros)... FLAGPTPV2
+	 *      0000              0000       0000 0010                    1
+	 *      PTPv2        Sync/Delay_Req      2                      PTPv2
+	 * Sync = 0000, Delay_Req = 0001 => MESSAGETYPE (value) = 000* (use 0 as *).
+	 * Ignore FLAGFIELD, DOMAINNUMBER. */
+	ret = mv3310_write_ptp_lut_reg(phydev, MV_V2_PTP_LUT_KEY_IG_BASE,
+				       0x00020001);
+	if (ret < 0)
+		return ret;
+
+	/* Set Ingress (RX) LUT Match Enable. This is mask. Set to 1 bit positions
+	 * from Ingress (RX) LUT Match Key above.
+	 * Check TRANSPORTSPECIFIC, MESSAGETYPE, VERSIONPTP and FLAGPTPV2:
+	 * TRANSPORTSPECIFIC   MESSAGETYPE   VERSIONPTP ...(zeros)... FLAGPTPV2
+	 *      1111              1110       0000 1111                    1
+	 *      PTPv2        Sync/Delay_Req      2                      PTPv2
+	 * Sync = 0000, Delay_Req = 0001 => MESSAGETYPE (mask) = 1110. */
+	ret = mv3310_write_ptp_lut_reg(phydev, MV_V2_PTP_LUT_KEY_IG_BASE + 8,
+				       0xfe0f0001);
+	if (ret < 0)
+		return ret;
+
+	return 0;
+}
+
+static int mv3310_ptp_set_lut_actions(struct phy_device *phydev, bool enable_tx,
+				      bool enable_rx)
+{
+	int ret;
+
+	/* Set Ingress (RX) LUT Action: INIPIGGYBACK
+	   Set Egress  (TX) LUT Action: UPDATERESIDENCE */
+	ret = mv3310_write_ptp_reg(phydev, MV_V2_PTP_LUT_ACTION_IG_BASE,
+				   enable_rx ? BIT(12) : 0);
+	if (ret < 0)
+		return ret;
+
+	return mv3310_write_ptp_reg(phydev, MV_V2_PTP_LUT_ACTION_EG_BASE,
```

Comment:
follow the same style for each call, anyway optimizer will generate the same seq of instructions